### DISCUSSION TITLE
Audacious and Thunar (folder contents sent from Thunar to Audacious, containing audio (& graphics) cause errors)

### DISCUSSION THREAD

**Discussion by jp9753** [audacious-media-player/audacious#88] [Q&A]
Linux Mint 20.3 (XFCE withThunar 4.16.8 )
audacious 3.10.1-1build1

I have used (the configure custom actions option), in Thunar, so a right-click (on any selected folder(s)), causes Thunar to send the folder(s) contents to audacious (for 'playing') (Thunar 'executes' /usr/bin/audacious %F)
but, I get (irritating) error dialog boxes, as there are also graphics (.jpg, etc) files in the folder(s), which causes audacious to issue the errors.

dare, I say VLC doesn't do this (it seems to simply ignore files it can't 'play')

I've tried options like:
/usr/bin/audacious %F/*.mp3 (works, but now audacious ignores .flac & .wav)
/usr/bin/audacious %F/*.mp3;*.flac;*.wav;*.ogg (works ((for mp3's), but .flac generates dialog box errors
/usr/bin/audacious %F/*.mp3 %F/*.flac %F/*.wav %F/*.ogg (audacious didn't like this)

turning on the 'audio files' (in Thunar configure custom actions) option doesn't work, because Thunar is interacting with the folder (not the files), also setting (Thunar's configure custom actions) file mask (with *.mp3;*.flac;*.wav;*.ogg) doesn't work for the same reasons.

using /usr/bin/audacious --enqueue %F, makes, no difference, placing (errror generating) jpg's (etc) in the playlist

so can any of the (input) plugins be configured, so non-playable files are filtered out

**Comment by radioactiveman** [comment]
With the settings from the [Xfce documentation](https://docs.xfce.org/xfce/thunar/custom-actions#playenqueue_music_files), I can open folders without issues, even if they contain files like images.
You should not need to specify file endings for the Thunar custom action.

```
Name: Play with Audacious
Command: audacious %F
File pattern: *
Appears if selection contains: Directories, Audio Files

Name: Enqueue in Audacious
Command: audacious -e %F
File pattern: *
Appears if selection contains: Directories, Audio Files
```

If FLAC files cannot be played, make sure you have the FLAC library installed (package named libflac or similar) and the Audacious FLAC plugin is enabled.

Last but not least, Audacious 3.10 was released in 2018. You may want to update to the [latest version](https://audacious-media-player.org/download).

**Reply by jp9753** [reply]
Thunar is executing /usr/bin/audacious %F
when playing a 'folder' audacious plays all the audio files in the folder with no trouble, but audacious generates an error (in its error dialog box) every time audacious encounters a non-audio file (ie a .jpg).

There may be a LARGE number of artwork jpg's residing in any of my album folders (or sub-folders), which makes audacious generates a large number of entries in its error dialog box. Each issued error (by audacious) moves the current focus from my current task.

Some of my box sets have audio files in multiple sub-folders with jpgs in the hundreds (big booklets or multiple booklets).

I have tried 'apt-get install audacious' (to upgrade to the latest available version), but, I have the latest Mint/Ubuntu repository version.
I already looked at the [[latest version](https://audacious-media-player.org/download)] page. I emailed Panda Jim about any known issues (with Mint) using his (Ubuntu) PPA, but haven't received a reply yet.
